src/shared/featureToggle.ts

```diff
@@ -44,7 +49,12 @@ export class FeatureToggle {
 
     private readonly maxFeatures: number = 5
 
-    public constructor(configuration: SettingsConfiguration, overrideKeys?: string[]) {
+    private static INSTANCE: FeatureToggle | undefined
+
+    public constructor(
```

Comment:
If you're going for a lazily loaded singleton, is there a reason that the constructor must also be public? Multiple instances could escape here.